### PR TITLE
Add binary sensor platform to Habitica

### PR DESCRIPTION
**Pull request by tr4nt0r**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/129613
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `binary_sensor` platform for the Habitica integration, allowing monitoring of binary states.
	- Added a **Pending quest invitation** sensor to notify users of any awaiting quest invitations.

- **Documentation**
	- Updated documentation to reflect the new capabilities and sensor descriptions, enhancing user awareness and experience.

### CODE REVIEW

File: source/_integrations/habitica.markdown [hunk 1]
```diff
@@ -58,7 +59,8 @@ The Habitca {% term integration %} enables you to monitor your adventurer's prog
 - **Rewards:** Displays the rewards that can be redeemed (for example, "1 task")
 - **Gems:** Shows the total number of gems currently owned by your Habitica character, used for purchasing items and customizations.
 - **Mystic hourglasses:** Displays the number of mystic hourglasses earned as a subscriber, which can be redeemed for exclusive items from past events.
-
+- **Pending quest invitation**: Indicates if you have an invitation to a quest awaiting your response.
```
Comment (c0ffeeca7) [inline source/_integrations/habitica.markdown:65]:
```suggestion
## Binary sensors

- **Pending quest invitation**: Indicates if you have an invitation to a quest awaiting your response.
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @tr4nt0r :+1:

**Comment by tr4nt0r** [comment]
@c0ffeeca7 parent hasn't been merged yet /o\

**Comment by c0ffeeca7** [comment]
> @c0ffeeca7 parent hasn't been merged yet /o\

I know... I was too trigger-happy

**Comment by frenck** [comment]
@tr4nt0r Instead of going down the path of reverting / re-opening a PR; let's try to get the parent PR merged 😉 

../Frenck

**Pull request merged**